aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/models/XRayTransactionState.java

```diff
@@ -100,4 +105,8 @@ public static void setServiceName(String inServiceName) {
     public static String getServiceName() {
         return serviceName;
     }
+
+    public static WeakConcurrentMap<PreparedStatement, String> getPreparedStatementMap() {
```

Comment:
Can we narrow down the returned interface to `Map`?

Reply:
Unfortunately not, the `WeakConcurrentMap` does not implement the `Map` interface to simplify/lighten its implementation per the README.

Reply:
I think two methods put and get allow making more narrow and could be better. Though API design doesn't matter so much in an agernt

Reply:
Good call, changed to just put/get methods